lib/Backend/BailOut.cpp

```diff
@@ -1215,7 +1215,7 @@ BailOutRecord::BailOutHelper(Js::JavascriptCallStackLayout * layout, Js::ScriptF
     // Clear the disable implicit call bit in case we bail from that region
     functionScriptContext->GetThreadContext()->ClearDisableImplicitFlags();
 
-    bool isInDebugMode = functionScriptContext->IsInDebugMode();
+    bool isInDebugMode = executeFunction->GetUtf8SourceInfo()->IsUtf8SourceInfoInDebugMode();
     AssertMsg(!isInDebugMode || Js::Configuration::Global.EnableJitInDebugMode(),
```

Comment:
nit: you could add IsInDebugMode to FunctionBody (which is essentially doing redirection to Utf8SourceInfo) that way it is easier to read in many places.